### ISSUE TITLE
hexo algolia failure for: TypeError: data.tags.map is not a function

### ISSUE THREAD

**Issue by AlanKeene** [theme-next/hexo-theme-next#212]
### I agree and want to create new issue 

- [x] Yes, I was on [Hexo Docs page](https://hexo.io/docs/), especially on [Templates](https://hexo.io/docs/templates.html), [Variables](https://hexo.io/docs/variables.html), [Helpers](https://hexo.io/docs/helpers.html) and [Troubleshooting](https://hexo.io/docs/troubleshooting.html).
- [x] Yes, I was on [NexT Documentation Site](http://theme-next.org/docs/).
- [x] And yes, I already searched for current [issues](https://github.com/theme-next/hexo-theme-next/issues?utf8=%E2%9C%93&q=is%3Aissue) and this is not help to me.

***

### Expected behavior 
I had updated my NexT-Mist from v5.1.4 to v6.1.0 followed the [https://github.com/theme-next/hexo-theme-next/blob/master/docs/UPDATE-FROM-5.1.X.md](https://github.com/theme-next/hexo-theme-next/blob/master/docs/UPDATE-FROM-5.1.X.md),
and installed the Algolia Search plugin followed the [https://github.com/theme-next/hexo-theme-next/blob/master/docs/ALGOLIA-SEARCH.md](https://github.com/theme-next/hexo-theme-next/blob/master/docs/ALGOLIA-SEARCH.md).
But when I run **"hexo algolia"** in Git Bash(hexo/), I got something wrong for: **"TypeError: data.tags.map is not a function"**

### Actual behavior 
Run **"hexo algolia"** failure.

### Steps to reproduce the behavior 
1. Followed the [https://github.com/theme-next/hexo-theme-next/blob/master/docs/UPDATE-FROM-5.1.X.md](https://github.com/theme-next/hexo-theme-next/blob/master/docs/UPDATE-FROM-5.1.X.md) to update the latest NexT version(v6.1.0).
2. Followed the [https://github.com/theme-next/hexo-theme-next/blob/master/docs/ALGOLIA-SEARCH.md](https://github.com/theme-next/hexo-theme-next/blob/master/docs/ALGOLIA-SEARCH.md) to install the Algolia plugin.
3. Run **“hexo algolia”** in Git Bash.

* Link to demo site with this issue: N/A
* Link(s) to source code or any usefull link(s): N/A

### Node.js and NPM Information

```
node -v
v8.10.0

npm -v
v5.6.0
```

### Package dependencies Information

```
$ cat package.json
{
  "name": "hexo-site",
  "version": "0.0.0",
  "private": true,
  "hexo": {
    "version": "3.6.0"
  },
  "dependencies": {
    "hexo": "^3.2.0",
    "hexo-algolia": "^1.2.4",
    "hexo-deployer-git": "^0.3.1",
    "hexo-generator-archive": "^0.1.4",
    "hexo-generator-category": "^0.1.3",
    "hexo-generator-feed": "^1.2.2",
    "hexo-generator-index": "^0.2.0",
    "hexo-generator-json-content": "^3.0.1",
    "hexo-generator-searchdb": "^1.0.8",
    "hexo-generator-tag": "^0.2.0",
    "hexo-helper-qrcode": "^1.0.2",
    "hexo-renderer-ejs": "^0.3.0",
    "hexo-renderer-less": "^0.2.0",
    "hexo-renderer-marked": "^0.3.0",
    "hexo-renderer-stylus": "^0.3.1",
    "hexo-server": "^0.2.0"
  }
}

```

#### Hexo version

```
$ hexo -v
hexo: 3.6.0
hexo-cli: 1.1.0
os: Windows_NT 10.0.16299 win32 x64
http_parser: 2.7.0
node: 8.10.0
v8: 6.2.414.50
uv: 1.19.1
zlib: 1.2.11
ares: 1.10.1-DEV
modules: 57
nghttp2: 1.25.0
openssl: 1.0.2n
icu: 60.1
unicode: 10.0
cldr: 32.0
tz: 2017c

```

#### Hexo Configuration

```yml
# my configuration in hexo/_config.yml is:
algolia:
  applicationID: 2L6xxxxxRKI3M
  apiKey: 251d016xxxxxxxxxbba54b3eaaff
  indexName: dev_blog
  chunkSize: 5000
```

### NexT Information

**NexT Version:**

- [x] Latest Master branch
- [ ] Latest Release version
- [ ] Old version

**NexT Scheme:**

- [ ] All schemes
- [ ] Muse
- [x] Mist
- [ ] Pisces
- [ ] Gemini

#### NexT Configuration:

```yml
# my configuration in next/_config.yml is:
algolia_search:
  enable: true
  hits:
    per_page: 10
  labels:
    input_placeholder: Search for Posts
    hits_empty: "We didn't find any results for the search: ${query}"
    hits_stats: "${hits} results found in ${time} ms"
```
And my hexo-algolia version is **"hexo-algolia@1.2.4"**

#### there are what I got when run **"hexo algolia"**:
```
$ hexo algolia
INFO  [Algolia] Testing HEXO_ALGOLIA_INDEXING_KEY permissions.
INFO  Start processing
FATAL Something's wrong. Maybe you can find the solution here: http://hexo.io/docs/troubleshooting.html
TypeError: data.tags.map is not a function
    at F:\MyBlog\Hexo\node_modules\hexo-algolia\lib\command.js:113:39
    at Array.map (<anonymous>)
    at F:\MyBlog\Hexo\node_modules\hexo-algolia\lib\command.js:99:37
    at <anonymous>
```
Has anyone face the same wrong?
Is this the Algolia's problem?

**Comment by sli1989** [comment]
Don't know why...

- add 'tags' in https://github.com/oncletom/hexo-algolia/blob/b04716a7ba70940094049a6816c767543404a424/lib/command.js#L13-L21?
- delete https://github.com/oncletom/hexo-algolia/blob/b04716a7ba70940094049a6816c767543404a424/lib/command.js#L112-L116?

**Comment by AlanKeene** [comment]
@sli1989 
- The source file had contains the part of these codes: [ https://github.com/oncletom/hexo-algolia/blob/b04716a7ba70940094049a6816c767543404a424/lib/command.js#L13-L2]( https://github.com/oncletom/hexo-algolia/blob/b04716a7ba70940094049a6816c767543404a424/lib/command.js#L13-L2).
- so we just delete [https://github.com/oncletom/hexo-algolia/blob/b04716a7ba70940094049a6816c767543404a424/lib/command.js#L112-L116](https://github.com/oncletom/hexo-algolia/blob/b04716a7ba70940094049a6816c767543404a424/lib/command.js#L112-L116) is OK.

And it is work for me.Thanks for your answer.

**Comment by ivan-nginx** [comment]
@sli1989 report bug to [Algolia repo](https://github.com/oncletom/hexo-algolia) maybe?

**Comment by sli1989** [comment]
@AlanKeene Can you provide the tags definition in front-matter of your md file.

**Comment by AlanKeene** [comment]
@sli1989 
My tags definition in front-matter of my md file is:
```
---
title: {{ title }}
date: {{ date }}
tags:
categories:
---
```
And the section in site's `_config.yml` is:
```
# Category & Tag
default_category: uncategorized
category_map:
tag_map:
```

**Comment by sli1989** [comment]
Bug fixed in [#34](https://github.com/oncletom/hexo-algolia/pull/34).
If issue solved, close it please.

**Comment by sli1989** [comment]
And as your [commented before](https://github.com/oncletom/hexo-algolia/issues/33#issuecomment-378998799), the definition of tags like below, it DID works.

```
tags: [tag1,tag2,tag3]
tags:
- PS3
- Games

tags:
  - tag1
  - tag2
```

the way blow is the way to add custom pages in `hexo\source\tags\index.md`. Of courese, need to make `tag_dir: tags` to active tags pages in site's `_config.yml`. They are irrelevant.

```
title: tags
date: 2015-10-20 06:49:50
type: "tags"
comments: false
```
